src/utils.cpp

```diff
@@ -68,6 +73,10 @@ void Dijkstra::dijkstra(const CouplingMap& couplingMap,
         newNode.prevCost = current->cost;
         newNode.pos      = to;
         newNode.containsCorrectEdge = correctEdge;
+        if (newNode.containsCorrectEdge && !current->containsCorrectEdge) {
+          newNode.cost -= reversalCost;
+          newNode.prevCost -= reversalCost;
+        }
```

Comment:
I might be wrong here, but can't it happen that this condition triggers multiple times?
For example on a path with 
`n0` -- correct -- `n1` -- wrong -- `n2` -- correct -- `n3` the condition would be triggered twice, right?

Reply:
I am quite certain that this condition only fires at most once per path, as any new node is always initialized with the same `containsCorrectEdge` as the previous node and only flipped to `true` given the right circumstances (and never back to `false`). In a way `containsCorrectEdge` encodes if the path up to and including this edge contains a forward-edge, instead of only the current edge. Therefore the flip from `false` to `true` only happens once.

In your example the condition would be triggered in the first edge and after that both `newNode.containsCorrectEdge` and `current->containsCorrectEdge` will both always be true (as `newNode.containsCorrectEdge` is initialized from `current->containsCorrectEdge` and can only be changed to true)

However, I see, how this might be unintuitive, and will add some comments explaining all of it as suggested by you.

Reply:
Ahhh 😌 now that you say it, I also see that. Maybe it was the GitHub diff that increased the confusion because I kept seeing the `false` initialization, but of course that was a deletion.
Then I am totally convinced that this is correct.
Adding a comment sounds like a good idea though!